source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -210,6 +211,71 @@ message FactBlockIsDead {
   uint32 block_id = 1;
 }
 
+message FactFunctionIsLivesafe {
+
+  // Records the fact that a function is guaranteed to be "livesafe", meaning
+  // that it will not make out-of-bounds accesses, does not contain reachable
+  // OpKill or OpUnreachable instructions, does not contain loops that will
+  // execute for large numbers of iterations, and only invokes other livesafe
+  // functions.
+
+  uint32 function_id = 1;
+}
+
+message AccessChainClampingInfo {
```

Comment:
This is so that we can use spirv-reduce to aggressively reduce a function, without it needing to maintain live-safety.  I'm imagining:

- use the shrinker to get rid of as many transformations as possible
- then apply spirv-reduce on each remaining donated function

The latter step will involve spirv-reduce trying a reduction, the function being re-added, being made livesafe in the process.